config/socket/socket.js

```diff
@@ -174,7 +175,10 @@ module.exports = function (io) {
     } else {
       game = gamesNeedingPlayers[0];
       allPlayers[socket.id] = true;
-
+      const isUserExist = game.players.map(user => user.userId === player.userId);
+      if (isUserExist[0] && player.userId !== 'unauthenticated') {
```

Comment:
Why are you comparing userId against unauthenticated, there should be some sort of property on user that can help you determine if the user is authenticated.

Reply:
In the player object, the parameters there are the username, userId, points, and avatar. Both authenticated and unauthenticated user has a username, the only difference is that userId of an unauthenticated user is the same.